Extensions/Mongo/Cosmos.DataTransfer.MongoExtension/MongoDataSourceExtension.cs

```diff
@@ -27,30 +28,95 @@ public async IAsyncEnumerable<IDataItem> ReadAsync(IConfiguration config, ILogge
 
             foreach (var collection in collectionNames)
             {
-                await foreach (var item in EnumerateCollectionAsync(context, collection, logger).WithCancellation(cancellationToken))
+                await foreach (var item in EnumerateCollectionAsync(context, collection, settings.Query, logger).WithCancellation(cancellationToken))
                 {
                     yield return item;
                 }
             }
         }
     }
 
-    public async IAsyncEnumerable<IDataItem> EnumerateCollectionAsync(Context context, string collectionName, ILogger logger)
+    public async IAsyncEnumerable<IDataItem> EnumerateCollectionAsync(Context context, string collectionName, string? query, ILogger logger)
     {
         logger.LogInformation("Reading collection '{Collection}'", collectionName);
         var collection = context.GetRepository<BsonDocument>(collectionName);
         int itemCount = 0;
-        foreach (var record in await Task.Run(() => collection.AsQueryable()))
+
+        IAsyncEnumerable<BsonDocument> documents;
+        
+        if (!string.IsNullOrWhiteSpace(query))
+        {
+            logger.LogInformation("Applying query filter to collection '{Collection}': {Query}", collectionName, query);
+            documents = GetQueryDocumentsAsync(collection, query, collectionName, logger);
+        }
+        else
+        {
+            // Use existing queryable approach when no filter is specified
+            documents = GetAllDocumentsAsync(collection);
+        }
+
+        await foreach (var record in documents)
         {
             yield return new MongoDataItem(record);
             itemCount++;
         }
+
         if (itemCount > 0)
             logger.LogInformation("Read {ItemCount} items from collection '{Collection}'", itemCount, collectionName);
         else
             logger.LogWarning("No items read from collection '{Collection}'", collectionName);
     }
 
+    private async IAsyncEnumerable<BsonDocument> GetAllDocumentsAsync(IRepository<BsonDocument> collection)
+    {
+        foreach (var record in await Task.Run(() => collection.AsQueryable()))
+        {
+            yield return record;
+        }
+    }
+
+    private async IAsyncEnumerable<BsonDocument> GetQueryDocumentsAsync(IRepository<BsonDocument> collection, string query, string collectionName, ILogger logger)
+    {
+        // Handle query as either a file path or direct JSON
+        string queryJson;
+        try
+        {
+            if (File.Exists(query))
+            {
+                logger.LogInformation("Reading query from file: {QueryFile}", query);
+                queryJson = await File.ReadAllTextAsync(query);
+            }
+            else
+            {
+                queryJson = query;
```

Comment:
Let's add a helpful logging line here that said why we think it's a query vs. a file.